### PR TITLE
Fix missing scrollbars on too-high menu-button and alt-F1 menus

### PR DESCRIPTION
**Pull request by lukefromdc**
This fixes part of  https://github.com/mate-desktop/mate-panel/issues/920

### CODE REVIEW

**Comment by raveit65** [comment]
I got a build warning.
```
panel-menu-button.c: In function 'panel_menu_button_popup_menu':
panel-menu-button.c:480:34: warning: passing argument 1 of 'gtk_menu_reposition' from incompatible pointer type [-Wincompatible-pointer-types]
  gtk_menu_reposition(button->priv->menu);
                      ~~~~~~~~~~~~^~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtklabel.h:34,
                 from /usr/include/gtk-3.0/gtk/gtkaccellabel.h:35,
                 from /usr/include/gtk-3.0/gtk/gtk.h:33,
                 from button-widget.h:4,
                 from panel-menu-button.h:28,
                 from panel-menu-button.c:27:
/usr/include/gtk-3.0/gtk/gtkmenu.h:176:49: note: expected 'GtkMenu *' {aka 'struct _GtkMenu *'} but argument is of type 'GtkWidget *' {aka 'struct _GtkWidget *'}
 void    gtk_menu_reposition    (GtkMenu        *menu);
                                 ~~~~~~~~~~~~~~~~^~~~
```

File: mate-panel/panel-menu-button.c [hunk 1]
```diff
@@ -476,6 +476,8 @@ panel_menu_button_popup_menu (PanelMenuButton *button,
 	                          GDK_GRAVITY_SOUTH_WEST,
 	                          GDK_GRAVITY_NORTH_WEST,
 	                          NULL);
+	/*Ensure scrollbars show up if vertical space is limited*/
```
Comment (raveit65) [inline mate-panel/panel-menu-button.c]: I would add a blank line before the comment.

**Comment by raveit65** [comment]
This fixes the build issue.
```
	/*Ensure scrollbars show up if vertical space is limited*/
	gtk_menu_reposition (GTK_MENU (button->priv->menu));
```

**Comment by lukefromdc** [comment]
I figured it would, and I am about to update this with both changes

**Comment by lukefromdc** [comment]
Just applied both changes asked for and force-pushed. Question: do we want to wait to see if GTK gets an upstream fix as per the discussion at and under
https://github.com/mate-desktop/mate-panel/issues/920#issuecomment-460218872
or merge this as soon as it can be tested? The issue did not appear until after GTK 3.24.3, when three new commits affected menus.
https://gitlab.gnome.org/GNOME/gtk/commit/00486efd51b29e234cfb180b814dd584805ebae5 would be the initial source of the problem,
https://gitlab.gnome.org/GNOME/gtk/commit/3e586a82e6cc563a84f7f16341c63743b0c62762
was no doubt supposed to ensure the menus were resized to fit but at least here didn't do it,
and 
https://gitlab.gnome.org/GNOME/gtk/commit/c35878ecf1e9ee960b72b213b73c15f92b64afe5
was to ensure the scroll arrow was correctly rendered but again this is not what we are getting.

I just tested master with today's GTK 3.24.5 and oh yeah, we still have the issue

**Comment by lukefromdc** [comment]
GTK 3 issue reported at
https://gitlab.gnome.org/GNOME/gtk/issues/1651

**Comment by raveit65** [comment]
> do we want to wait to see if GTK gets an upstream fix as per the discussion at and under
[#920 (comment)](https://github.com/mate-desktop/mate-panel/issues/920#issuecomment-460218872)
or merge this as soon as it can be tested?

I would wait a bit, let see how gtk upstream react and if they do a new release with a fix.
On the other side we can always revert you fix when it is fixed by upstream.
Keep in mind that it take some time when a fix from our side will find a way in distros and a fix by gtk+ is already there.

**Comment by lukefromdc** [comment]
Albert just reproduced this on a gnome-terminal menu bar, which ought to get the GNOME devs working on this

**Comment by lukefromdc** [comment]
We now have a PR for GTK3 that fixes this right:
https://gitlab.gnome.org/GNOME/gtk/merge_requests/565
so unless that gets rejected we won't need this PR here, which will become unneeded code after the GTK fix gets merged. Holding this open in case https://gitlab.gnome.org/GNOME/gtk/merge_requests/565 does not get merged for any reason

**Comment by lukefromdc** [comment]
gitlab.gnome.org/GNOME/gtk/commit/1d4eac211c09624d29b9309e2b92173f7477a9b7
has been merged and fixes this, so closing this PR which was at best a partial workaround and now becomes unneeded code

**Comment by raveit65** [comment]
@lukefromdc 
Can branch dev-all be deleted?

**Comment by lukefromdc** [comment]
yes, it can go

**Pull request closed without merging**